packages/gcf-utils/test/bot-request.ts

```diff
@@ -0,0 +1,154 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {describe, it} from 'mocha';
+import assert from 'assert';
+import * as express from 'express';
+import {parseBotRequest, TriggerType} from '../src/bot-request';
+
+describe('parseBotRequest', () => {
+  describe('with all headers', () => {
+    const headers = {
+      'x-github-event': 'issues',
```

Comment:
I would randomize the case a bit in the headers, to confirm that request doesn't care.

Reply:
Looks like the header getter is case insensitive, but the normalization happens sometime in the building of the request object.

Reply:
The header accessors are now referencing the upper camel-cased versions of the headers and the test is using all lowercase now.